pkg/hubble/relay/pool/types/client.go

```diff
@@ -36,3 +46,23 @@ type ClientConnBuilder interface {
 	// optionally, hostname.
 	ClientConn(target, hostname string) (ClientConn, error)
 }
+
+// WaitForReady blocks until the provided connection becomes Ready or timeout is reached.
+func WaitForReady(conn ClientConn, timeout time.Duration) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		s := conn.GetState()
+		if s == connectivity.Idle {
+			conn.Connect()
```

Comment:
the grpc ClientConn has a `WaitForStateChange` which seems like it could be used instead potentially?

Reply:
I actually wanted to use it, but I saw it was only used in DialContext, so I avoided it in case it would disappear in the future.